### PR TITLE
test(agent): cover dynamic skill tools in AgentTool subagents

### LINKED ISSUE #4656: [BUG] AgentTool模式，子agent的skill如果关联了mcp工具，mcp工具无法调用
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

子Agent的skill关联了mcp工具，在AgentLlmNode类的buildChatClientRequestSpec方法中将工具放到了config，config.context().put(RunnableConfig.DYNAMIC_TOOL_CALLBACKS_METADATA_KEY, modelRequest.getDynamicToolCallbacks());
但在AgentToolNode类的resolveFromConfigMetadata方法中获取不到工具，config.context().get(RunnableConfig.DYNAMIC_TOOL_CALLBACKS_METADATA_KEY)取到了空，此类工具最终调用报错：No ToolCallback found for tool name:  xxxx

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Environment

```markdown
Spring AI Alibaba version(s):
```

### Debug logs

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by March-77**
### Describe what this PR does / why we need it

Adds deterministic regression coverage for the multi-agent scenario reported in #4656. The test exercises a streaming parent agent that invokes an AgentTool sub-agent; the sub-agent reads a skill, receives a grouped dynamic ToolCallback, and executes that callback successfully.

The reported failure does not reproduce on the current `main` branch. The missing piece is automated coverage for this exact configuration boundary, so this PR protects the working RunnableConfig propagation behavior from regressions without changing production code.

### Does this pull request fix one issue?

Fixes #4656

### Describe how you did it

- Added scripted parent and child ChatModel fixtures so the test is deterministic and does not require an external model or MCP server.
- Streamed the parent ReactAgent and delegated through `AgentTool.create(subAgent)`.
- Configured the child with `SkillsAgentHook.groupedTools`.
- Made the child call `read_skill`, then call the dynamically activated `record_result` ToolCallback.
- Asserted that the grouped tool executes with the expected value.

User impact: maintainers now have a focused regression test for dynamic skill-tool propagation across streaming AgentTool boundaries. Runtime behavior and public APIs are unchanged.

### Describe how to verify it

Run:

```bash
JAVA_HOME=/opt/homebrew/opt/openjdk PATH=/opt/homebrew/opt/openjdk/bin:$PATH ./mvnw -pl spring-ai-alibaba-agent-framework -Dtest=AgentToolConfigPropagationTest test
```

Result: BUILD SUCCESS; 3 tests run, 0 failures, 0 errors, 0 skipped. Checkstyle and Spotless also passed as part of the command.

### Special notes for reviews

This is intentionally test-only because the exact reporter-provided path works on current `main`. The test uses the existing `grouped-tools-test` fixture and a generic ToolCallback, matching the issue details. Compatibility risk is limited to test execution; there is no production or API change.

### CODE REVIEW

File: spring-ai-alibaba-agent-framework/src/test/java/com/alibaba/cloud/ai/graph/agent/AgentToolConfigPropagationTest.java [hunk 1]
```diff
@@ -103,6 +110,45 @@ void shouldIncludeContextInInvocationFailureMessage() {
...
+				.hooks(List.of(skillsHook))
+				.build();
+		ReactAgent parentAgent = ReactAgent.builder()
+				.name("streaming_parent_agent")
+				.model(new ParentToolCallingChatModel())
+				.tools(AgentTool.create(subAgent))
+				.build();
+
+		RunnableConfig parentConfig = RunnableConfig.builder().threadId("parent-thread").build();
+		assertNotNull(parentAgent.stream("delegate this task", parentConfig).blockLast());
```
[the remark this answers is not in the document]
Comment (March-77) [reply spring-ai-alibaba-agent-framework/src/test/java/com/alibaba/cloud/ai/graph/agent/AgentToolConfigPropagationTest.java]: Fixed in `1bd01ca86`. The streaming parent test now installs a config-capture hook on the child and asserts the child receives the parent-derived thread id plus parent business metadata, in addition to verifying the dynamically activated tool executes. All 3 config propagation tests pass and Checkstyle is clean.

**Review by oss-sentinel-ai — COMMENTED** [review]
### Summary
Reviewed the changes in this PR (154 lines changed). The changes appear to be well-structured.

### Observations
- Changes span 154 lines
- Please ensure all edge cases are covered
- Verify test coverage for the modified code paths

---
*Automated review by github-manager-bot*

**Comment by March-77** [comment]
@yuluo-yx 已重新核对当前 head、最新 main 变化、mergeability、CI、reviews 和 review threads；现有评审意见均已验证并清理，未发现需要追加代码修改的问题。烦请在方便时进行维护者审查，谢谢。
